src/flight.rs

```diff
@@ -115,7 +115,10 @@ impl SqlFlightClient {
         &mut self,
         query: &str,
     ) -> std::result::Result<FlightRecordBatchStream, Box<dyn Error>> {
-        self.authenticate().await?;
+        let api_key = self.api_key.clone();
+        if let Some(api_key) = api_key {
```

Comment:
```suggestion
        if let Some(api_key) = api_key.as_ref() {
```

Can avoid a clone here

Reply:
compiler complains:

![CleanShot 2024-04-26 at 18 14 17@2x](https://github.com/spiceai/spice-rs/assets/827338/b60fbea8-97d3-4fd6-85f4-7c4bd5ad64d1)